includes/connect/class-wc-stripe-connect-api.php

```diff
@@ -0,0 +1,244 @@
+			$business_data['first_name']    = $current_user->user_firstname;
+			$business_data['last_name']     = $current_user->user_lastname;
+			$business_data['phone']         = '';
+			$business_data['currency']      = get_woocommerce_currency();
+
+			$wc_countries = WC()->countries;
+
+			if ( method_exists( $wc_countries, 'get_base_address' ) ) {
+				$business_data['country']        = $wc_countries->get_base_country();
+				$business_data['street_address'] = $wc_countries->get_base_address();
+				$business_data['city']           = $wc_countries->get_base_city();
+				$business_data['state']          = $wc_countries->get_base_state();
+				$business_data['zip']            = $wc_countries->get_base_postcode();
+			} else {
+				$base_location                   = wc_get_base_location();
+				$business_data['country']        = $base_location['country'];
+				$business_data['street_address'] = '';
+				$business_data['city']           = '';
+				$business_data['state']          = $base_location['state'];
+				$business_data['zip']            = '';
+			}
+
+			$request = array(
+				'returnUrl'    => $return_url,
+				'businessData' => $business_data,
+			);
+
+			return $this->request( 'POST', '/stripe/oauth-init', $request );
+		}
+
+		/**
+		 * Send request to Connect Server for Stripe keys
+		 *
+		 * @param  string $code OAuth server code.
+		 *
+		 * @return array
+		 */
+		public function get_stripe_oauth_keys( $code ) {
+
+			$request = array( 'code' => $code );
+
+			return $this->request( 'POST', '/stripe/oauth-keys', $request );
+		}
+
+		/**
+		 * General OAuth request method.
+		 *
+		 * @param string $method request method.
+		 * @param string $path   path for request.
+		 * @param array  $body   request body.
+		 *
+		 * @return array|WP_Error
+		 */
+		protected function request( $method, $path, $body = array() ) {
+
+			if ( ! is_array( $body ) ) {
+				return new WP_Error(
+					'request_body_should_be_array',
+					__( 'Unable to send request to WooCommerce Connect server. Body must be an array.', 'woocommerce-gateway-stripe' )
+				);
+			}
+
+			$url = trailingslashit( WOOCOMMERCE_CONNECT_SERVER_URL );
+			$url = apply_filters( 'wc_connect_server_url', $url );
+			$url = trailingslashit( $url ) . ltrim( $path, '/' );
+
+			// Add useful system information to requests that contain bodies.
+			if ( in_array( $method, array( 'POST', 'PUT' ), true ) ) {
+				$body = $this->request_body( $body );
+				$body = wp_json_encode( apply_filters( 'wc_connect_api_client_body', $body ) );
+
+				if ( ! $body ) {
+					return new WP_Error(
+						'unable_to_json_encode_body',
+						__( 'Unable to encode body for request to WooCommerce Connect server.', 'woocommerce-gateway-stripe' )
+					);
+				}
+			}
+
+			$headers = $this->request_headers();
+			if ( is_wp_error( $headers ) ) {
+				return $headers;
+			}
+
+			$http_timeout = 60; // 1 minute
+			if ( function_exists( 'wc_set_time_limit' ) ) {
```

Comment:
Probably don't need to check for this? wc_set_time_limit has been around since 2.6 and this plugin requires 3.0

Reply:
Good call. Removed this check in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1267/commits/c286591a251eda4439044105aaff161011b1fba0